src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -219,11 +221,11 @@ public void Clear()
         /// <returns>true if item is found in the <see cref="ReadOnlyCollectionBuilder{T}"/>; otherwise, false.</returns>
         public bool Contains(T item)
         {
-            if ((Object)item == null)
+            if ((object)item == null)
```

Comment:
Yes, it treats it as `if(false)` and treats the whole branch as dead code it doesn't need to jit.